plasmapy/particles/particle_collections.py

```diff
@@ -321,16 +343,61 @@ def sort(self, key: Callable = None, reverse: bool = False):
         """
         Sort the |ParticleList| in-place.
 
-        For more information, refer to the documentation for `list.sort`.
+        Parameters
+        ----------
+        key : callable
+            A function of one argument that is used to extract a
+            comparison key for each item in the |ParticleList|.
+
+        reverse : `bool`, default: `False`
+            If `True`, the items in the |ParticleList| are sorted as if
+            each comparison were reversed.
+
+        Raises
+        ------
+        TypeError
+            If ``key`` is not provided.
+
+        See Also
+        --------
+        list.sort
+        sorted
+
+        Examples
+        --------
+        To sort a |ParticleList| by atomic number, set ``key`` to
+        |atomic_number|.
+
+        >>> from plasmapy.particles import ParticleList, atomic_number
+        >>> elements = ParticleList(["Li", "H", "He"])
+        >>> elements.sort(key=atomic_number)
+        >>> print(elements)
+        ParticleList(['H', 'He', 'Li'])
+
+        We can also create a function that we then pass to ``key``. In
```

Comment:
I don't quite understand this example yet - how do the three quantities coming from this function define the sorting? Does it sort based off of the first one and then break ties with the next one? I think that could be said a tad more explicitly.